### PR TITLE
Drop legacy and make components pure JSR330

### PR DESCRIPTION
**Pull request by cstamas**
Drop legacy and complete plexus container, as it is not needed anymore.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,17 +61,62 @@
...
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest</artifactId>
+      <version>2.2</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>${slf4jVersion}</version>
```
Comment (michael-o) [inline pom.xml:87]: test scope

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:87]: fixed

File: src/main/java/org/codehaus/plexus/components/io/filemappers/DefaultFileMapper.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+
+import javax.inject.Named;
+
+/**
+ * Alias for {@link IdentityMapper}
+ */
+@Named
+public class DefaultFileMapper extends IdentityMapper
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/plexus/components/io/filemappers/DefaultFileMapper.java:25]: Does singleton is missing?

File: src/main/java/org/codehaus/plexus/components/io/fileselectors/AllFilesFileSelector.java [hunk 3]
```diff
@@ -17,10 +17,12 @@
  */
 
 import javax.annotation.Nonnull;
+import javax.inject.Named;
 
 /**
  * The default file selector: Selects all files.
  */
+@Named( AllFilesFileSelector.ROLE_HINT )
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/plexus/components/io/fileselectors/AllFilesFileSelector.java:25]: This was singleton.

File: src/main/java/org/codehaus/plexus/components/io/fileselectors/IncludeExcludeFileSelector.java [hunk 4]
```diff
@@ -24,11 +24,13 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import javax.inject.Named;
 
 /**
  * This file selector uses a set of patterns for including/excluding
  * files.
  */
+@Named( IncludeExcludeFileSelector.ROLE_HINT )
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/plexus/components/io/fileselectors/IncludeExcludeFileSelector.java:33]: This was singleton.

Comment (cstamas) [reply src/main/java/org/codehaus/plexus/components/io/fileselectors/IncludeExcludeFileSelector.java:33]: This was wrong then: this class is stateful, it cannot be singleton.

**Review by michael-o — APPROVED** [review]

**Review by michael-o — APPROVED** [review]

**Review by olamy — APPROVED** [review]

**Pull request merged**